pexv2/pd.go

```diff
@@ -22,6 +34,98 @@ type InputDescriptor struct {
 	Constraints Constraints `json:"constraints"`
 }
 
+type tokenizedField struct {
+	token string
+	path  string
+}
+
+// SelectCredentials selects vcJWTs based on the constraints defined in the input descriptor
+func (ind InputDescriptor) SelectCredentials(vcJWTs []string) ([]string, error) {
+	jsonSchema := JSONSchema{
+		Schema:     "http://json-schema.org/draft-07/schema#",
+		Type:       "object",
+		Properties: make(map[string]Filter, len(ind.Constraints.Fields)),
+		Required:   make([]string, 0, len(ind.Constraints.Fields)),
+	}
+
+	tokenizedFields := make([]tokenizedField, 0, len(ind.Constraints.Fields))
+	tokens := make([]string, len(ind.Constraints.Fields))
+
+	for i, field := range ind.Constraints.Fields {
+		token := strconv.FormatInt(time.Now().UnixNano(), 10)
+		tokens[i] = token
+
+		for _, path := range field.Path {
+			tf := tokenizedField{token: token, path: path}
+			tokenizedFields = append(tokenizedFields, tf)
+		}
+
+		if field.Filter != nil {
+			jsonSchema.AddProperty(token, *field.Filter)
```

Comment:
do we not also have to add the token as a `required` field in json schema while building the schema here? i recall the lib i was using i had to do that.

Reply:
The tests passed without it. good catch though! forgot to add to `required`. Added: https://github.com/TBD54566975/web5-go/pull/144/commits/47f285c54364ce76100ba8333f3b97f104829589